spec/system/admin/reports_spec.rb

```diff
@@ -352,7 +352,24 @@
       visit admin_reports_path
 
       click_link 'All products'
-      run_failed_report(Reporting::Reports::ProductsAndInventory::AllProducts)
+      report = Reporting::Reports::ProductsAndInventory::AllProducts
+
+      click_on "Go"
+
+      allow(report).to receive(:new).and_raise(StandardError, 'Provoked error for testing')
+      perform_enqueued_jobs(only: ReportJob)
+
+      expect(page).not_to have_selector ".loading"
+      expect(page).to have_button "Go", disabled: false
+      expect(page).to have_content 'This report failed. It may be too big to process. ' \
+                                   'We will look into it but please let us know ' \
+                                   'if the problem persists.'
+
+      # Admin shoulb be able to make some changes and retry
+      sleep(1)
+      allow(report).to receive(:new).and_call_original
+
+      run_report
```

Comment:
I think we should check the report is actually displayed on the page. Something like `expect(page).to have_content "Supplier" ` should be enough.

Reply:
I just added that line.